include/ygm/container/experimental/detail/maptrix_impl.hpp

```diff
@@ -0,0 +1,143 @@
+
+  /* For all is expected to be const on csc. */
+  template <typename Function>
+  void for_all(Function fn) {
+    m_csc.for_all(fn);
+  }
+
+  template <typename Function>
+  void for_all_row(Function fn) {
+    m_csr.for_all_row(fn);
+  }
+
+  template <typename Function>
+  void for_all_col(Function fn) {
+    m_csc.for_all_col(fn);
+  }
+
+  template <typename... VisitorArgs>
+  void print_all(std::ostream& os, VisitorArgs const&... args) {
+    ((os << args), ...);
+  }
+
+  template <typename Visitor, typename... VisitorArgs>
+  void async_visit_if_exists(const key_type &row, const key_type &col, 
+          Visitor visitor, const VisitorArgs &...args) {
+    m_csr.async_visit_if_exists(row, col, visitor, std::forward<const VisitorArgs>(args)...);
+    m_csc.async_visit_if_exists(row, col, visitor, std::forward<const VisitorArgs>(args)...);
+  }
+
+  template <typename Visitor, typename... VisitorArgs>
+  void async_visit_col_mutate(const key_type &col, Visitor visitor,
+                             const VisitorArgs&... args) {
+    auto &m_map     = m_csc.csc();
+    auto &inner_map = m_map.find(col)->second;
+    for (auto itr = inner_map.begin(); itr != inner_map.end(); ++itr) {
+      key_type row  = itr->first;
+      m_csr.async_visit_if_exists(row, col, visitor, std::forward<const VisitorArgs>(args)...);
+      m_csc.async_visit_if_exists(row, col, visitor, std::forward<const VisitorArgs>(args)...);
+    }
+  }
+
+  template <typename Visitor, typename... VisitorArgs>
+  void async_visit_row_const(const key_type &row, Visitor visitor,
+                             const VisitorArgs &...args) {
+    m_csr.async_visit_row_const(row, visitor, std::forward<const VisitorArgs>(args)...);
+  }
+
+  template <typename Visitor, typename... VisitorArgs>
+  void async_visit_col_const(const key_type &col, Visitor visitor,
+                             const VisitorArgs &...args) {
+    m_csc.async_visit_col_const(col, visitor, std::forward<const VisitorArgs>(args)...);
+  }
+
+  template <typename Visitor, typename... VisitorArgs>
+  void async_insert_if_missing_else_visit(const key_type &row, const key_type &col, const value_type &value, 
+                                Visitor visitor, const VisitorArgs&... args) {
+    m_csr.async_insert_if_missing_else_visit(row, col, value, visitor, std::forward<const VisitorArgs>(args)...);
+    m_csc.async_insert_if_missing_else_visit(row, col, value, visitor, std::forward<const VisitorArgs>(args)...);
+  }
+
+  typename ygm::ygm_ptr<self_type> get_ygm_ptr() const { return pthis; }
+
+  void local_clear() { 
+    m_csr.clear(); 
+    m_csc.clear(); 
+  }
+
+  void swap(self_type &s) {
+    m_csr.swap(s.csr);
+    m_csc.swap(s.csc);
+  }
+
+ protected:
+  maptrix_impl() = delete;
+
+  value_type                                          m_default_value;
+  csr_impl                                            m_csr;      
+  csc_impl                                            m_csc; 
```

Comment:
Change the 'csr' and 'csc' naming to 'row_view' and 'column_view'

Reply:
Yep, will do.